mlx/backend/accelerate/primitives.cpp

```diff
@@ -255,6 +257,96 @@ void AsType::eval_cpu(const std::vector<array>& inputs, array& out) {
   eval(inputs, out);
 }
 
+void BluesteinFFTSetup::eval_cpu(
+    const std::vector<array>& inputs,
+    std::vector<array>& outputs) {
+  // We need to calculate the Bluestein twiddle factors
+  // in double precision for the overall numerical stability
+  // of Bluestein's FFT algorithm to be acceptable.
+  //
+  // MLX doesn't currently support float64, so instead we
+  // manually implement the required operations using accelerate.
+  //
+  // In numpy:
+  // w_k = np.exp(-1j * np.pi / N * (np.arange(-N + 1, N) ** 2))
+  // w_q = np.fft.fft(1/w_k)
+  // return w_k, w_q
```

Comment:
What do you think of section IV.E of https://mc.stanford.edu/cgi-bin/images/7/75/SC08_FFT_on_GPUs.pdf . Would it solve our problem here to avoid double precision arithmetic?

Reply:
Fig 6 is very promising :-)

Reply:
This looks nice! I simplified the double precision part a bit so I think I'm going to keep it for now since it's not really an accuracy or performance bottleneck. Happy to revisit in the future though.